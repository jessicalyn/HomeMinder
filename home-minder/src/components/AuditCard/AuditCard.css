.audit-card {
  border: solid .1rem black;
  margin: 1rem auto;
  padding: 1rem;
  width: 200px;
  height: 100px;
}

h3 {
  margin-top: 0
}