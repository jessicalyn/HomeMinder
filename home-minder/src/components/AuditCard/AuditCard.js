import React from 'react';
import { NavLink } from 'react-router-dom';
import './AuditCard.css'

const AuditCard = ({ room, name, id }) => (
  <div className="audit-card">
    <h3>{ room } { name }</h3>
    <NavLink to={`/item/${id}`}>Schedule</NavLink>
  </div>
)

export default AuditCard