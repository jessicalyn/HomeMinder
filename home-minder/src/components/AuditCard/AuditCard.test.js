import React from 'react'
import AuditCard from './AuditCard';
import { shallow } from 'enzyme'

describe('AuditCard', () => {
  let wrapper
  const item = {
    id: 1,
    name: "Refrigerator Water Filter",
    room: "Kitchen"
  }

  it('should match the snapshot with data passed in', () => {
    wrapper = shallow(<AuditCard {...item} />)
    expect(wrapper).toMatchSnapshot()
  })
})