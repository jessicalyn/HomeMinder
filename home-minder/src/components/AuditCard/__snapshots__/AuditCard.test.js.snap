// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuditCard should match the snapshot with data passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AuditCard
    id={1}
    name="Refrigerator Water Filter"
    room="Kitchen"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Kitchen
           
          Refrigerator Water Filter
        </h3>,
        <NavLink
          to="/item/1"
        >
          Schedule
        </NavLink>,
      ],
      "className": "audit-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Kitchen",
            " ",
            "Refrigerator Water Filter",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Kitchen",
          " ",
          "Refrigerator Water Filter",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Schedule",
          "to": "/item/1",
        },
        "ref": null,
        "rendered": "Schedule",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Kitchen
             
            Refrigerator Water Filter
          </h3>,
          <NavLink
            to="/item/1"
          >
            Schedule
          </NavLink>,
        ],
        "className": "audit-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Kitchen",
              " ",
              "Refrigerator Water Filter",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Kitchen",
            " ",
            "Refrigerator Water Filter",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Schedule",
            "to": "/item/1",
          },
          "ref": null,
          "rendered": "Schedule",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
